packages/teuchos/core/src/Teuchos_ScalarTraits.cpp

```diff
@@ -119,7 +120,7 @@ bool Teuchos::operator&&(const qd_real &a, const qd_real &b) {
 #ifndef __sun
 // This is an intentional computation of NaN.
 namespace Teuchos {
-const float  flt_nan = +returnFloatZero()/returnFloatZero();
-const double dbl_nan = +returnDoubleZero()/returnDoubleZero();
+  const float  flt_nan = std::numeric_limits<float>::signaling_NaN();
```

Comment:
@bathmatt wrote:
> If you enable FPE checking too early teuchos throws an exception, this fixes that..

It looks like this creates a signaling NaN.  Wouldn't you want a quiet NaN (`std::numeric_limits<float>::quiet_NaN()`) instead?

Reply:
Well, if one uses it should it signal?  I'm fine with quiet_NaN since we don't use it anywhere,   But if I was to use flt_nan as a default to make sure my code died if I used uninitialized memory I'd be angry that it wsa a quiet one.

Reply:
Also, it was a signaling nan before.

Reply:
@bathmatt Try it and see if it passes the tests?